### PR TITLE
WIP: feat: support EntryPlugin filename function

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

Allow EntryPlugin filename configuration to be a function.

fix https://github.com/web-infra-dev/rspack/issues/5971

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: crates/rspack_core/src/chunk_group.rs [hunk 1]
```diff
@@ -356,15 +355,22 @@ impl EntryRuntime {
...
   pub async_chunks: Option<bool>,
   pub public_path: Option<PublicPath>,
   pub base_uri: Option<String>,
-  pub filename: Option<FilenameTemplate>,
+  #[derivative(
+    Hash = "ignore",
+    PartialEq = "ignore",
+    PartialOrd = "ignore",
+    Ord = "ignore"
+  )]
```
Comment (9aoy) [inline crates/rspack_core/src/chunk_group.rs:372]: @ahabhgk  help~ I'm not sure if ignored is possible here

Comment (SyMind) [reply crates/rspack_core/src/chunk_group.rs:372]: The Hash, PartialEq, Eq, PartialOrd, and Ord traits can potentially be removed now.
These were previously used for testing in Rust.

**Comment by phyyou** [comment]
Hi, @9aoy. Are you still working on this PR?

**Comment by 9aoy** [comment]
> Hi, @9aoy. Are you still working on this PR?

Waiting for somebody to remove the Hash, PartialEq, Eq, PartialOrd, and Ord traits. I tried to remove it, but it failed due to too many associated functions 😂 .

**Comment by phyyou** [comment]
> > Hi, @9aoy. Are you still working on this PR?
> 
> Waiting for somebody to remove the Hash, PartialEq, Eq, PartialOrd, and Ord traits. I tried to remove it, but it failed due to too many associated functions 😂 .

I see. I would like to give it a try. Although I'm not familiar with this, I'm eager to try. Any tips or guide would be greatly appreciated!

**Comment by xc2** [comment]
@9aoy @phyyou 
Please update the branch from main branch, and `Filename` will work without removing or ignoring `Hash`, `PartialEq`, `Eq`, `PartialOrd`, and `Ord` for now.

**Comment by 9aoy** [comment]
This PR is a bit outdated, and I will open a new one.

**Pull request closed without merging**